adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -990,26 +991,29 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 	action.Route.ClusterSpecifier = headerBasedClusterSpecifier
 	logger.LoggerOasparser.Debug("added header based cluster")
 
-	if (prodRouteConfig != nil && prodRouteConfig.RetryConfig != nil) ||
-		(sandRouteConfig != nil && sandRouteConfig.RetryConfig != nil) {
-		// Retry configs are always added via headers. This is to update the
-		// default retry back-off base interval, which cannot be updated via headers.
-		retryConfig := config.Envoy.Upstream.Retry
-		commonRetryPolicy := &routev3.RetryPolicy{
-			RetryOn: retryPolicyRetriableStatusCodes,
-			NumRetries: &wrapperspb.UInt32Value{
-				Value: 0,
-				// If not set to 0, default value 1 will be
-				// applied to both prod and sandbox even if they are not set.
-			},
-			RetriableStatusCodes: retryConfig.StatusCodes,
-			RetryBackOff: &routev3.RetryPolicy_RetryBackOff{
-				BaseInterval: &durationpb.Duration{
-					Nanos: int32(retryConfig.BaseIntervalInMillis) * 1000,
+	if os.Getenv("ROUTER_CONNECTION_FAILURE_RETRY_ENABLED") != "" {
+		if prodRouteConfig != nil || sandRouteConfig != nil {
+			// Retry configs are always added via headers. This is to update the
+			// default retry back-off base interval, which cannot be updated via headers.
+			retryConfig := config.Envoy.Upstream.Retry
+			maxInterval := retryConfig.MaxInterval
+			if retryConfig.MaxInterval < retryConfig.BaseInterval {
+				maxInterval = retryConfig.BaseInterval
+			}
+			commonRetryPolicy := &routev3.RetryPolicy{
+				RetryOn: retryOnConnectFailures,
+				NumRetries: &wrapperspb.UInt32Value{
+					Value: retryConfig.MaxRetryCount,
+					// If not set to 0, default value 1 will be
+					// applied to both prod and sandbox even if they are not set.
 				},
-			},
+				RetryBackOff: &routev3.RetryPolicy_RetryBackOff{
+					BaseInterval: durationpb.New(retryConfig.BaseInterval),
+					MaxInterval:  durationpb.New(maxInterval),
+				},
+			}
+			action.Route.RetryPolicy = commonRetryPolicy
```

Comment:
We can set this config in the Vhost level too right, instead of each route level?

Reply:
yeah good point.
Made the change.